### PR TITLE
Bump trivy to v0.54.1 and fix breaking changes

### PR DESCRIPTION
**Pull request by romulets**
### Summary of your changes

- Bump trivy to v0.54.1
- Fix breaking change introduced by  zap replacement by slog (https://github.com/aquasecurity/trivy/pull/6466) 
  - I find it weird that trivy uses the [default logger](https://github.com/aquasecurity/trivy/blob/main/pkg/log/logger.go) for everything. But that's why I used `SetDefault` from `github.com/aquasecurity/trivy/pkg/log`, which overrides slog default logger
- Fix breaking change introduced by renaming `vuln` flags by `pckg` flags (https://github.com/aquasecurity/trivy/pull/7104)
- Fix breaking change introduced by adoption of `name.Reference` on DBRepositories (https://github.com/aquasecurity/trivy/pull/6410)

### Related Issues

- Removes CVE https://github.com/elastic/security/issues/2143

### CODE REVIEW

**Review by moukoublen — APPROVED** [review]

File: internal/flavors/vulnerability.go [hunk 1]
```diff
@@ -53,9 +54,8 @@ func NewVulnerability(b *beat.Beat, cfg *agentconfig.C) (beat.Beater, error) {
 	// Override trivy's logger
 	scanLog := zap.New(log.Core()).Sugar()
 	dlog.SetLogger(scanLog)
-	flog.SetLogger(scanLog)
 	xlog.SetLogger(scanLog)
-	tlog.Logger = scanLog
+	tlog.SetDefault(slog.New(zapslog.NewHandler(log.Core(), nil)))
```
Comment (romulets) [inline internal/flavors/vulnerability.go:58]: Fix breaking change introduced by zap replacement by slog (https://github.com/aquasecurity/trivy/pull/6466)
I find it weird that trivy uses the [default logger](https://github.com/aquasecurity/trivy/blob/main/pkg/log/logger.go) for everything. But that's why I used SetDefault from github.com/aquasecurity/trivy/pkg/log, which overrides slog default logger

File: internal/vulnerability/runner.go [hunk 2]
```diff
@@ -48,17 +49,17 @@ func NewVulnerabilityRunner(log *logp.Logger) (VulnerabilityRunner, error) {
 			Quiet:   false,
 			Debug:   true,
 		},
-		VulnerabilityOptions: flag.VulnerabilityOptions{
-			VulnType: []string{trivy_types.VulnTypeOS, trivy_types.VulnTypeLibrary},
+		PackageOptions: flag.PackageOptions{
+			PkgTypes: []string{trivy_types.PkgTypeOS, trivy_types.PkgTypeLibrary},
```
Comment (romulets) [inline internal/vulnerability/runner.go:53]: Fix breaking change introduced by renaming vuln flags by pckg flags (https://github.com/aquasecurity/trivy/pull/7104)

File: internal/vulnerability/scanner.go [hunk 3]
```diff
@@ -117,8 +117,8 @@ func (f VulnerabilityScanner) scan(ctx context.Context, snap ec2.EBSSnapshot) {
 			Quiet:   false,
 			Debug:   true,
 		},
-		VulnerabilityOptions: flag.VulnerabilityOptions{
-			VulnType: []string{trivy_types.VulnTypeOS, trivy_types.VulnTypeLibrary},
+		PackageOptions: flag.PackageOptions{
+			PkgTypes: []string{trivy_types.PkgTypeOS, trivy_types.PkgTypeLibrary},
```
Comment (romulets) [inline internal/vulnerability/scanner.go:121]: Fix breaking change introduced by renaming vuln flags by pckg flags (https://github.com/aquasecurity/trivy/pull/7104)

File: internal/vulnerability/runner.go [hunk 4]
```diff
@@ -48,17 +49,17 @@ func NewVulnerabilityRunner(log *logp.Logger) (VulnerabilityRunner, error) {
...
 		ScanOptions: flag.ScanOptions{
 			Scanners: []trivy_types.Scanner{trivy_types.VulnerabilityScanner},
 			RekorURL: "https://rekor.sigstore.dev",
 		},
 		DBOptions: flag.DBOptions{
 			NoProgress:       true,
-			DBRepository:     "ghcr.io/aquasecurity/trivy-db",
-			JavaDBRepository: "ghcr.io/aquasecurity/trivy-java-db",
+			DBRepository:     name.MustParseReference("ghcr.io/aquasecurity/trivy-db"),
+			JavaDBRepository: name.MustParseReference("ghcr.io/aquasecurity/trivy-java-db"),
```
Comment (romulets) [inline internal/vulnerability/runner.go:62]: Fix breaking change introduced by adoption of name.Reference on DBRepositories (https://github.com/aquasecurity/trivy/pull/6410)

File: internal/vulnerability/scanner.go [hunk 5]
```diff
@@ -117,8 +118,9 @@ func (f VulnerabilityScanner) scan(ctx context.Context, snap ec2.EBSSnapshot) {
 			Quiet:   false,
 			Debug:   true,
 		},
-		VulnerabilityOptions: flag.VulnerabilityOptions{
-			VulnType: []string{trivy_types.VulnTypeOS, trivy_types.VulnTypeLibrary},
+		PackageOptions: flag.PackageOptions{
+			PkgTypes:         []string{trivy_types.PkgTypeOS, trivy_types.PkgTypeLibrary},
+			PkgRelationships: ftypes.Relationships,
```
Comment (romulets) [inline internal/vulnerability/scanner.go:123]: Package relationship has been introduced on https://github.com/aquasecurity/trivy/pull/7237 and using CLI there are default relationship values (all). Using the library directly there is no default value and having 0 relationships and retrieving 0 finginds. Therefore on this commit we added the default value with all relationships

File: internal/vulnerability/runner.go [hunk 6]
```diff
@@ -48,17 +50,18 @@ func NewVulnerabilityRunner(log *logp.Logger) (VulnerabilityRunner, error) {
 			Quiet:   false,
 			Debug:   true,
 		},
-		VulnerabilityOptions: flag.VulnerabilityOptions{
-			VulnType: []string{trivy_types.VulnTypeOS, trivy_types.VulnTypeLibrary},
+		PackageOptions: flag.PackageOptions{
+			PkgTypes:         []string{trivy_types.PkgTypeOS, trivy_types.PkgTypeLibrary},
+			PkgRelationships: ftypes.Relationships,
```
Comment (romulets) [inline internal/vulnerability/runner.go:55]: Package relationship has been introduced on https://github.com/aquasecurity/trivy/pull/7237 and using CLI there are default relationship values (all). Using the library directly there is no default value and having 0 relationships and retrieving 0 finginds. Therefore on this commit we added the default value with all relationships

**Pull request merged**